spot-contracts/contracts/vaults/RolloverVault.sol

```diff
@@ -185,6 +186,25 @@ contract RolloverVault is
 
     /// @inheritdoc IVault
     function recover() public override nonReentrant whenNotPaused {
+        // Redeem perp for tranches
+        uint256 perpBalance = perp.balanceOf(address(this));
+        if (perpBalance > 0) {
+            (IERC20Upgradeable[] memory tranchesRedeemed, ) = perp.computeRedemptionAmts(perpBalance);
```

Comment:
Looking through the lens of efficiency...

- We don't really care about computing the redemption amounts. We really only care about getting the list of reserve assets.
- Both computeRedemptionAmts and redeem call updateState() on perp, even though we only need to do it once.

I wonder, would it be better if redeem() just returned the assets itself? Seems like it would be useful for clients, as this one shows. Then we could make just the one call.

We also may end up syncing a particular tranche more than once, if there are any in common between the perp redeem and the deployed set. The multiple syncs would probably require building up a list of assets to sync then doing them at the end, but I'm not sure the gas savings would pencil out

Reply:
Agreed we can just make all 3 functions, ie) deposit, redeem and rollover return the full data.